pkgs/development/libraries/highfive/default.nix

```
@@ -0,0 +1,43 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+, boost
+, eigen
+, hdf5
+}:
+
+stdenv.mkDerivation rec {
+  pname = "highfive";
+  version = "2.2";
+
+  src = fetchFromGitHub {
+    owner = "BlueBrain";
+    repo = "HighFive";
+    rev = "4c70d818ed18231563fe49ff197d1c41054be592";
+    sha256 = "02xy3c2ix3nw8109aw75ixj651knzc5rjqwqrxximm4hzwx09frk";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ boost eigen hdf5 ];
+
+  inherit (hdf5) mpiSupport;
+
```

This is likely what you want
```suggestion

  passthru = {
    inherit (hdf5) mpiSupport mpi;
  };
```